aiida_quantumespresso/calculations/epw.py

```diff
@@ -0,0 +1,302 @@
+# -*- coding: utf-8 -*-
+"""Plugin to create a Quantum Espresso epw.x input file."""
+from __future__ import absolute_import
+
+import os
+import six
+from six.moves import range
+
+from aiida import orm
+from aiida.common import datastructures, exceptions
+from aiida_quantumespresso.calculations import _lowercase_dict, _uppercase_dict
+from aiida_quantumespresso.utils.convert import convert_input_to_namelist_entry
+from .base import CalcJob
+
+
+class EpwCalculation(CalcJob):
+    """`CalcJob` implementation for the epw.x code of Quantum ESPRESSO."""
+
+    # Keywords that cannot be set by the user but will be set by the plugin
+    _blocked_keywords = [('INPUTEPW', 'outdir'), ('INPUTEPW', 'iverbosity'), ('INPUTEPW', 'prefix'),
+                         ('INPUTEPW', 'dvscf_dir'), ('INPUTEPW', 'amass'), ('INPUTEPW', 'nq1'), ('INPUTEPW', 'nq2'),
+                         ('INPUTEPW', 'nq3'), ('INPUTEPW', 'nk1'), ('INPUTEPW', 'nk2'), ('INPUTEPW', 'nk3')]
+
+    _use_kpoints = True
+
+    _compulsory_namelists = ['INPUTEPW']
+
+    # Default input and output files
+    _PREFIX = 'aiida'
+    _DEFAULT_INPUT_FILE = 'aiida.in'
+    _DEFAULT_OUTPUT_FILE = 'aiida.out'
+    _OUTPUT_XML_TENSOR_FILE_NAME = 'tensors.xml'
+    _OUTPUT_SUBFOLDER = './out/'
+    _SAVE_PREFIX = '/save/'
+    _FOLDER_SAVE = 'save'
+    _FOLDER_DYNAMICAL_MATRIX = 'DYN_MAT'
+
+    # Not using symlink in pw to allow multiple nscf to run on top of the same scf
+    _default_symlink_usage = False
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super(EpwCalculation, cls).define(spec)
+        spec.input('metadata.options.input_filename', valid_type=six.string_types, default=cls._DEFAULT_INPUT_FILE)
+        spec.input('metadata.options.output_filename', valid_type=six.string_types, default=cls._DEFAULT_OUTPUT_FILE)
+        spec.input('metadata.options.withmpi', valid_type=bool, default=True)
+        spec.input('kpoints', valid_type=orm.KpointsData, help='coarse kpoint mesh')
+        spec.input('qpoints', valid_type=orm.KpointsData, help='coarse qpoint mesh')
+        spec.input('kfpoints', valid_type=orm.KpointsData, help='fine kpoint mesh')
+        spec.input('qfpoints', valid_type=orm.KpointsData, help='fine qpoint mesh')
+        spec.input('qibz', valid_type=orm.ArrayData, help='qpoint mesh in IBZ (must be same as prior PhCalculation')
```

Comment:
Didn't we determine last time that this could be determined from the `PhCalculation`? I think we checked that it is in fact already parsed and so does not have to be passed in as a separate input node.

If you remove it here, make sure to also update the test and the submission script